### DISCUSSION TITLE
Is Gatsby still committed to open source?

### DISCUSSION THREAD

**Discussion by jstormail2** [gatsbyjs/gatsby#36488] [Help]
At first blush, this may seem like a silly question.  I’m opening this issue on GitHub after all.

I’m re-evaluating NextJS, GatsbyJS, RemixJS, and other frameworks for a large upcoming project.

Previously, I switched to NextJS and have been using it for the past few years.  But, I want to give GatsbyJS a fair shot.

However, everywhere I look I see Gatsby Cloud, and I am having trouble seeing whether features ever made it back to the open source project:
- https://www.gatsbyjs.com/blog/re-introducing-gatsby-a-reactive-site-generator/
- https://www.gatsbyjs.com/gatsby-4/

I would just like to confirm features like Deferred Static Generation (DSG) and Server-Side Rendering (SSR) are *only* available in Gatsby Cloud.  

I remember incremental builds from v3 were made open-source, but I cannot find references to DSG or SSR outside of Gatsby Cloud.  As far as I can tell, self-hosted builds only appear to run Gatsby with incremental builds.

---

For comparison: 

- NextJS has open-sourced Incremental Static Regeneration (ISR) in both their cloud and self-hosted deployments.  
- NextJS supports multiple image cdn providers.  While theirs works out of the box, there are still self-hosted and other options.

There are many more features than the two examples above, but the only feature from NextJS not commonly implemented in OSS/self-hosted solutions is middleware, as it depends on the hosted provider.

---

I could go on, but the feeling I have with Gatsby v4 is that the only viable path forward is vendor lock-in with Gatsby Cloud.  

Please correct me if I’m wrong!  I want to 💜 Gatsby.  I used it happily in the past, but I feel like Gatsby Cloud is taking a path that I cannot follow.

---

I’m not saying I’m against Gatsby Cloud and would never use it.  I just want to confirm that these important features are open source and not behind vendor lock-in.  For the previous NextJS project, portions of the site are self-hosted and portions of the site are on Vercel.  Usually, hosting is a mix of services anyway.

**Comment by Khaledgarbaya** [comment]
Hi @jstormail2,

This is Khaled from the framework team at Gatsby, thanks for this great question.
Yes, Gatsby is committed to open source and will always be. We work in public with our community members to help shape the future of Gatsby through the [RFC](https://github.com/gatsbyjs/gatsby/discussions/categories/rfc) process.

For features like DSG and SSR they are available to self-host using a plugin like [gatsby-plugin-fastify](https://www.gatsbyjs.com/plugins/gatsby-plugin-fastify/) so they are not gated or vendor locked but implementation can be done in a lot of ways so it does not make sense to have just one OSS implementation.

Besides DSG and SSR we added other features like the [Head API](https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-head/) and the [Script component](https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-script/).

Stay tuned for Gatsby 5 with more amazing features, it's around the corner.
